packages/opentelemetry/src/utils/enhanceDscWithOpenTelemetryRootSpanName.ts

```diff
@@ -9,20 +10,31 @@ import { spanHasName } from './spanTypes';
  */
 export function enhanceDscWithOpenTelemetryRootSpanName(client: Client): void {
   client.on('createDsc', (dsc, rootSpan) => {
+    if (!rootSpan) {
+      return;
+    }
+
     // We want to overwrite the transaction on the DSC that is created by default in core
     // The reason for this is that we want to infer the span name, not use the initial one
     // Otherwise, we'll get names like "GET" instead of e.g. "GET /foo"
     // `parseSpanDescription` takes the attributes of the span into account for the name
     // This mutates the passed-in DSC
-    if (rootSpan) {
-      const jsonSpan = spanToJSON(rootSpan);
-      const attributes = jsonSpan.data || {};
-      const source = attributes[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE];
 
-      const { description } = spanHasName(rootSpan) ? parseSpanDescription(rootSpan) : { description: undefined };
-      if (source !== 'url' && description) {
-        dsc.transaction = description;
-      }
+    const jsonSpan = spanToJSON(rootSpan);
+    const attributes = jsonSpan.data || {};
+    const source = attributes[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE];
+
+    const { description } = spanHasName(rootSpan) ? parseSpanDescription(rootSpan) : { description: undefined };
+    if (source !== 'url' && description) {
+      dsc.transaction = description;
+    }
+
+    // Also ensure sampling decision is correctly inferred
+    // In core, we use `spanIsSampled`, which just looks at the trace flags
+    // but in OTEL, we use a slightly more complex logic to be able to differntiate between unsampled and deferred sampling
+    if (hasTracingEnabled()) {
```

Comment:
this is _kind of_ not exactly what the name implies this does, but I think it's OK to add this here too. Otherwise, we'll have to export another, new method here and use this in the Node SDK `initOtel`, which seems not really worth it here 🤔

Reply:
I think we should rename `hasTracingEnabled` anyway, which we can do during v9